include/alpaka/block/shared/OffloadUseBuiltInSharedMem.hpp

```diff
@@ -0,0 +1,26 @@
+/* Copyright 2022 Jeffrey Kelling
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+namespace alpaka
+{
+#ifndef ALPAKA_OFFLOAD_USE_BUILTIN_SHARED_MEM
+#    define ALPAKA_OFFLOAD_USE_BUILTIN_SHARED_MEM OFF
+#endif
+    enum class OffloadUseBuiltInSharedMem : char
+    {
+        OFF, //! Do not use built-in shared memory facilites, use BlockSharedMemDynMember
+        DYN_FIXED, //! Use built-in shared memory, allocate fixed space for dyn smem
+        DYN_ALLOC //! Use built-in shared memory, use runtime allocation API
+    };
+
+    constexpr OffloadUseBuiltInSharedMem OFFLOAD_USE_BUILTIN_SHARED_MEM
+        = OffloadUseBuiltInSharedMem::ALPAKA_OFFLOAD_USE_BUILTIN_SHARED_MEM;
+} // namespace alpaka
```

Comment:
How about something like this?

```suggestion
namespace alpaka
{
    struct OffloadBuiltInSharedMemOff {};
    struct OffloadBuiltInSharedMemFixed {};
    struct OffloadBuiltInSharedMemAlloc {};
#ifdef ALPAKA_OFFLOAD_BUILTIN_SHARED_MEM_FIXED
    using OffloadBuiltInSharedMem = OffloadBuiltInSharedMemFixed;
#elif defined(ALPAKA_OFFLOAD_BUILTIN_SHARED_MEM_ALLOC)
    using OffloadBuiltInSharedMem = OffloadBuiltInSharedMemAlloc;
#else
    using OffloadBuiltInSharedMem = OffloadBuiltInSharedMemOff;
#endif
} // namespace alpaka
```

This would interact with the other suggested changes below and allow for user extensibility.

Reply:
Looks good.

I see on issue, though: While this allows the user to add new variants on which they can also specialize `AccOmp5BlockSharedMem<>` below, they cannot make `AccOmp5` use any custom strategy, because the complie-time switch above always defines `alpaka::OffloadBuiltInSharedMem`. I guess, to fully allow extensibility, one would need to add a flag like `ALPAKA_OFFLOAD_BUILTIN_SHARED_MEM_USER` which would leave `OffloadBuiltInSharedMem` undefined, allowing the user to define it, before including `AccOmp5.hpp`.